### PR TITLE
upgrade Scala.js versions (add 1.0.0 support)

### PR DESCRIPTION
**Pull request by SethTisue**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -13,8 +13,6 @@
 import sbt._
 import org.scalajs.sbtplugin.ScalaJSPlugin
 import org.scalajs.sbtplugin.ScalaJSPlugin.autoImport._
-// shadow sbt-scalajs' crossProject and CrossType until Scala.js 1.0.0 is released
-import sbtcrossproject.{ crossProject, CrossType }
```
Comment (sjrd) [inline build.sbt]: If you cross-build for 0.6.x, you'll still need this. (it's a no-op with 1.x.)

Comment (SethTisue) [reply build.sbt]: amended, thanks

**Comment by SethTisue** [comment]
```
[error] org.scalajs.core.ir.IRVersionNotSupportedException: Failed to deserialize info of file compiled with Scala.js 1.0 (supported: 0.6.0, 0.6.13, 0.6.3, 0.6.17, 0.6.4, 0.6.6, 0.6.14, 0.6.8, 0.6.30, 0.6.29, 0.6.5, 0.6.15): /home/travis/build/scala/nanotest-strawman/js/target/scala-2.11/test-classes/example/sourcecodetest/Apply.sjsir
```

**Comment by SethTisue** [comment]
reproducible locally. `env SCALAJS_VERSION=0.6.32 sbt "verifyJS/test"` fails, same thing succeeds with 1.0.0. maybe something in the build isn't being overridden properly when we set `SCALAJS_VERSION`, so we end up with some mix of 0.6 and 1.0 stuff?

(rather than mess with it, I'm pretty tempted to just drop Scala.js 0.6 support, which I had considered doing anyway...)

**Comment by SethTisue** [comment]
ah but it succeeds if I `;clean;verifyJS/test` instead of just `verifyJS/test`. I think I remember seeing somebody else's build that they had to clean in between

**Comment by SethTisue** [comment]
@eed3si9n this is ready for review/merge. can you make sure the `.travis.yml` changes make sense to you? It appears to me that as it was before, we were double-testing Scala.js stuff.

**Comment by eed3si9n** [comment]
> It appears to me that as it was before, we were double-testing Scala.js stuff.

I think we were testing Scala.JS 0.6 in the first three lines.

**Comment by eed3si9n** [comment]
Back published 0.2.0 against Scala JS 1.

**Comment by SethTisue** [comment]
> I think we were testing Scala.JS 0.6 in the first three lines.

ah, I see. well, I think it's nicer the new way, with the JVM and JS stuff separate.

**Pull request merged**